packages/mobx/src/types/observablemap.ts

```diff
@@ -306,14 +306,14 @@ export class ObservableMap<K = any, V = any>
     }
 
     /** Merge another object into this object, returns this. */
-    merge(other: ObservableMap<K, V> | IKeyValueMap<V> | any): ObservableMap<K, V> {
+    merge(other?: IObservableMapInitialValues<K, V>): ObservableMap<K, V> {
```

Comment:
The `undefined` type was added because method `merge` may accept `undefined` according to ObservabelMap's constructor: https://github.com/mobxjs/mobx/blob/main/packages/mobx/src/types/observablemap.ts#L101

Merging Map with undefined doesn't make much sense, but adding `if` type guard would be a BC break. That's why I decided not to affect runtime